libs/render_delegate/node_graph.cpp

```diff
@@ -43,454 +43,153 @@
-                TfToken vartoken;
-                const auto paramIt = inputNode->parameters.find(str::t_varname);
-                if (paramIt != inputNode->parameters.end()) {
-                    if (paramIt->second.IsHolding<TfToken>()) {
-                        vartoken = paramIt->second.UncheckedGet<TfToken>();
-                    } else if (paramIt->second.IsHolding<std::string>()) {
-                        vartoken = TfToken(paramIt->second.UncheckedGet<std::string>());
-                    }
-                }
-                if (!vartoken.IsEmpty()) {
-                    // We want to delete the connection and avoid relying on the uvcoords attribute.
-                    // However we need to keep the inputId, otherwise we won't be able to find
-                    // the entry point to the shader network.
-                    relationship.outputId = SdfPath();
-                    // if the primvar being used is "uv" or "st", then we can just rely on defaults.
-                    // Otherwise we need to set the attribute uvset of the output image node
-                    if (vartoken != str::t_uv && vartoken != str::t_st) {
-                        outputNode->parameters[str::t_uvset] = VtValue(vartoken);
-                    }
-                } 
-            }
-        }
-        // We test if a texture is connected to the opacity, and if it is we invert it as later on we remap
-        // the opacity to the transparency. It is a workaround and might be problematic if the texture is
-        // connected to other nodes which don't expect an inverted texture.
-        if (relationship.outputName == str::t_opacity) {
-            for (auto& material : network.nodes) {
-                if (material.path == relationship.inputId && material.identifier == str::t_UsdUVTexture) {
-                    auto scaleFound = material.parameters.find(str::t_scale);
-                    auto biasFound = material.parameters.find(str::t_bias);
-                    GfVec4f oriBias = (biasFound == material.parameters.end()) ? GfVec4f(0.f) : biasFound->second.Get<GfVec4f>();
-                    GfVec4f oriScale = (scaleFound == material.parameters.end()) ? GfVec4f(1.f) : scaleFound->second.Get<GfVec4f>();
-                    material.parameters[str::t_scale] = VtValue(oriScale*-1.f);
-                    material.parameters[str::t_bias] = VtValue(GfVec4f(1.f)-oriBias);
-                }
-            }
-        }
-    }
-
-    for (auto& material : network.nodes) {
-        const auto remapIt = _NodeRemapFuncs().find(material.identifier);
-        if (remapIt == _NodeRemapFuncs().end()) {
-            continue;
-        }
-
-        HydraMaterialNetworkEditContext editCtx(network, material);
-        remapIt->second(&editCtx);
+    // Ensure all AtNodes created for this node graph are properly deleted
+    for (const auto& node : _nodes) {
+        if (node.second)
+            AiNodeDestroy(node.second);
```

Comment:
it would be great to make sure _nodes doesn't contain invalid data, would it make sense to clear it ? or assign nullptr to each node ? (node.second = nullptr)

Reply:
but this is in the class destructor, so the whole vector will be gone after that

Reply:
didn't see it was in the destructor, sloppy review ...